src/category_theory/equivalence.lean

```diff
@@ -1,156 +1,256 @@
+  have := (ε.app $ F.obj X).hom_inv_id, dsimp at this, rw [this], clear this,
+  rw [id_comp], have := (F.map_iso $ η.app X).hom_inv_id, dsimp at this, rw [this]
+end
+
 end
 
+protected definition mk (F : C ⥤ D) (G : D ⥤ C)
+  (η : 𝟭 C ≅ F ⋙ G) (ε : G ⋙ F ≅ 𝟭 D) : C ≌ D :=
+⟨F, G, adjointify_η η ε, ε, adjointify_ε_η η ε⟩
+
+omit 𝒟
+@[refl] def refl : C ≌ C := equivalence.mk (𝟭 C) (𝟭 C) (iso.refl _) (iso.refl _)
+include 𝒟
+
+@[symm] def symm (e : C ≌ D) : D ≌ C :=
+⟨e.inverse, e.functor, e.counit_iso.symm, e.unit_iso.symm, e.inverse_counit_inv_comp⟩
+
 variables {E : Sort u₃} [ℰ : category.{v₃} E]
 include ℰ
 
-@[simp] private def effe_iso_id (e : C ≌ D) (f : D ≌ E) (X : C) :
-  (e.inverse).obj ((f.inverse).obj ((f.functor).obj ((e.functor).obj X))) ≅ X :=
+@[simp] private def id_iso_effe (e : C ≌ D) (f : D ≌ E) (X : C) :
+  X ≅ (e.inverse).obj ((f.inverse).obj ((f.functor).obj ((e.functor).obj X))) :=
 calc
-  (e.inverse).obj ((f.inverse).obj ((f.functor).obj ((e.functor).obj X)))
-    ≅ (e.inverse).obj ((e.functor).obj X) : e.inverse.map_iso (nat_iso.app f.fun_inv_id _)
-... ≅ X                                   : nat_iso.app e.fun_inv_id _
+ X ≅ (e.inverse).obj ((e.functor).obj X) : e.unit_iso.app _
+... ≅ _                                  : e.inverse.map_iso (f.unit_iso.app _)
 
 @[simp] private def feef_iso_id (e : C ≌ D) (f : D ≌ E) (X : E) :
   (f.functor).obj ((e.functor).obj ((e.inverse).obj ((f.inverse).obj X))) ≅ X :=
 calc
-  (f.functor).obj ((e.functor).obj ((e.inverse).obj ((f.inverse).obj X)))
-    ≅ (f.functor).obj ((f.inverse).obj X) : f.functor.map_iso (nat_iso.app e.inv_fun_id _)
-... ≅ X                                   : nat_iso.app f.inv_fun_id _
+  _ ≅ (f.functor).obj ((f.inverse).obj X) : f.functor.map_iso (e.counit_iso.app _)
+... ≅ X                                   : f.counit_iso.app _
 
 @[trans] def trans (e : C ≌ D) (f : D ≌ E) : C ≌ E :=
-{ functor := e.functor ⋙ f.functor,
-  inverse := f.inverse ⋙ e.inverse,
-  fun_inv_id' := nat_iso.of_components (effe_iso_id e f)
-  begin
-    /- `tidy` says -/
-    intros X Y f_1, dsimp at *, simp at *, dsimp at *,
-    /- `rewrite_search` says -/
-    slice_lhs 3 4 { erw [is_iso.hom_inv_id] },
-    erw [category.id_comp, is_iso.hom_inv_id, category.comp_id],
-  end,
-  inv_fun_id' := nat_iso.of_components (feef_iso_id e f)
-  begin
-    /- `tidy` says -/
-    intros X Y f_1, dsimp at *, simp at *, dsimp at *,
-    /- `rewrite_search` says -/
-    slice_lhs 3 4 { erw [is_iso.hom_inv_id] },
-    erw [category.id_comp, is_iso.hom_inv_id, category.comp_id]
-  end
-}
+begin
+  apply equivalence.mk (e.functor ⋙ f.functor) (f.inverse ⋙ e.inverse),
+  { fapply of_components, exact id_iso_effe e f,
+    intros X Y g, dsimp at *, simp at *, dsimp at *,
+    slice_rhs 2 3 { erw [is_iso.hom_inv_id] }, rw [id_comp],
+    slice_rhs 1 2 { erw [is_iso.hom_inv_id] }, rw [id_comp, assoc] },
+  fapply of_components, exact feef_iso_id e f,
```

Comment:
I would tend to use `{ ... }` also for the last goal when it is parallel to an earlier goal, and try to make the two proofs look similar.